### PR TITLE
fix: storybook flakiness and minor typescript issues

### LINKED ISSUE #5526: Flaky Test on `Codebox` for Storybook
### The Issue

The Codebox component currently holds a flaky test that often passes and often does not. It also causes PRs that run `test:storybook:snapshot` often to update this flaky Storybook test with unrelated changes.

#### Cause

This is mainly caused due to `Prismsjs` being client-side and modifying the DOM-tree on-the-fly for the Codebox. This is the current approach instead of running Prismjs, setting the content on a state, and then injecting through a `dangerouslyInnerHtml` React option.

The new approach does indeed modify the contents, and it is a side-effect out of the bounds of React.

### Solution

Since this change on the DOM-tree is caused by Prismjs, we can wait for the `tabindex` field to be present. (Wait until `pre[tabindex="0"]` renders)

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR introduces a fix for the Storybook flakiness by adding a 1s timeout on Playwright. This is a "sad" fix since we cannot customise each test (at least, I did not find any docs on Storybook's `test-runner` to configure each test individually (e.g. add steps before render or after render). The 1-second delay is a global way of ensuring that the Snapshots are always of the latest rendered state. It would be super inefficient if we have any component that'd take more than 1 second to render.

This PR also does minor ESLint fixes, updates the `package-lock.json` (outdated about the `package.json`), and some TypeScript complaints. (cc @bmuenzenmeyer, we might want to add a step on the `pull-request.yml` checks to run `tsc` to validate typescript).

Finally it also adds `--coverage` on jest for a future PR adding coverage reports.

## Validation

This PR should fix Storybook's flakiness.

## Related Issues

Fixes https://github.com/nodejs/nodejs.org/issues/5526

### CODE REVIEW

File: .storybook/test-runner.ts [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 
+const STORYBOOK_ELEMENT_ID = '[data-test-id="story-root"]';
+
 const config: TestRunnerConfig = {
-  postRender: async (page, context) => {
-    // Gather the page HTML inner content for a DOM HTML Snapshot
-    const rootElementId = '[data-test-id="story-root"]';
-    const rootElement = await page.locator(rootElementId);
+  postRender: async page => {
+    // We wait the page for loading for at least one second
```
Comment (bmuenzenmeyer) [inline .storybook/test-runner.ts]:
```suggestion
    // We wait for the page to load for at least one second
```

**Review by bmuenzenmeyer — APPROVED** [review]
My valiation:

I ran this 10 times locally, worked everytime.
I reverted the 1 second pause, it failed immediately.

**Comment by bmuenzenmeyer** [comment]
@nodejs/website requesting fast track, this is causing other PRs to flap green/red

File: .storybook/test-runner.ts (hunk 1 above)
Comment (ovflowd) [reply .storybook/test-runner.ts]: I can still apply this suggestion if you’d like 👀

**Comment by ovflowd** [comment]
I can’t self approve my own PRs, but +1 anyways from my side 😝

**Pull request merged**